File: apps/frontend/src/types/content.ts
export type SubscriptionTier = 0 | 1 | 2 | 3;
export type ContentType = 'Movie' | 'Series' | 'Documentary' | 'LiveEvent' | 'TVChannel';
export type ProcessingStatus = 'Pending' | 'Processing' | 'Ready' | 'Failed';

export interface ContentSummary {
  id: string;
  title: string;
  slug: string;
  thumbnailUrl: string | null;
  backdropUrl: string | null;
  type: ContentType;
  releaseYear: number;
  durationMinutes: number;
  requiredTier: SubscriptionTier;
  averageRating: number;
  isFeatured: boolean;
  isTrending: boolean;
}

export interface ContentDetail extends ContentSummary {
  description: string;
  genres: Genre[];
  seasons: Season[];
  videoAssets: VideoAsset[];
  subtitles: Subtitle[];
}

export interface Genre {
  id: string;
  name: string;
  slug: string;
}

export interface Season {
  id: string;
  seasonNumber: number;
  title: string;
  episodes: Episode[];
}

export interface Episode {
  id: string;
  episodeNumber: number;
  title: string;
  description: string;
  durationMinutes: number;
  thumbnailUrl: string | null;
}

export interface VideoAsset {
  id: string;
  quality: string;
  status: ProcessingStatus;
  hlsManifestPath: string | null;
}

export interface Subtitle {
  languageCode: string;
  languageName: string;
  fileUrl: string;
}

export interface StreamInfo {
  assetId: string;
  masterM3u8Url: string;
  qualities: string[];
  subtitles: Subtitle[];
}

File: apps/backend/src/StreamingPlatform.Domain/Entities/Content.cs
using StreamingPlatform.Domain.Common;
using StreamingPlatform.Domain.Enums;

namespace StreamingPlatform.Domain.Entities;

public class Content : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? ThumbnailUrl { get; set; }
    public string? BackdropUrl { get; set; }
    public int ReleaseYear { get; set; }
    public int DurationMinutes { get; set; }
    public ContentType Type { get; set; }
    public SubscriptionTier RequiredTier { get; set; }
    public bool IsPublished { get; set; }
    public bool IsFeatured { get; set; }
    public bool IsTrending { get; set; }
    public decimal AverageRating { get; set; }
    public ICollection<ContentGenre> ContentGenres { get; set; } = new List<ContentGenre>();
    public ICollection<Season> Seasons { get; set; } = new List<Season>();
    public ICollection<VideoAsset> VideoAssets { get; set; } = new List<VideoAsset>();
    public ICollection<Subtitle> Subtitles { get; set; } = new List<Subtitle>();
    public ICollection<WatchlistItem> WatchlistItems { get; set; } = new List<WatchlistItem>();
    public ICollection<WatchProgress> WatchProgresses { get; set; } = new List<WatchProgress>();
}
